crates/orderbook/openapi.yml

```diff
@@ -540,32 +530,113 @@ components:
+          $ref: "#/components/schemas/Signature"
+        from:
+          description: |
+            If set, the backend enforces that this address matches what is decoded as the signer of
+            the signature. This helps catch errors with invalid signature encodings as the backend
+            might otherwise silently work with an unexpected address that for example does not have
+            any balance.
+          allOf:
+            - $ref: "#/components/schemas/Address"
+          nullable: true
+        quoteId:
+          description: |
+            Orders can optionally include a quote ID. This way the order can be linked to a quote
+            and enable providing more metadata when analyzing order slippage.
+          type: integer
+          nullable: true
+        appData:
+          description: |
+            This field comes in two forms for backward compatibility. The first form (app data hash) will eventually stop being accepted.
+          anyOf:
+            - $ref: "#/components/schemas/AppDataHash"
+            - title: Full App Data
               description: |
-                Orders can optionally include a quote ID. This way the order can be linked to a quote
-                and enable providing more metadata when analyzing order slippage.
-              type: integer
-              nullable: true
-          required:
-            - signingScheme
-            - signature
+                Short:
+
+                If you do not care about app data, set this field to `\"{}\"` and make sure that the order you signed for this request had its app data field set to `0xb48d38f93eaa084033fc5970bf96e559c33c4cdc07d889ab00b4d63f9590739d`.
+
+                Long:
+
+                A string encoding a json object like `"{\"hello\":\"world\"}"`.
+
+                This field determines the smart contract order's app data field, which is assumed to be set to the keccak-256 hash of the utf-8 encoded bytes of this string. You must ensure that the signature that is part of this request indeed signed a smart contract order with the app data field set appropriately. If this isn't the case, signature verification will fail. For easier debugging you can additionally set the `appDataHash` field.
+
+                This field must be the encoding of a valid json object. The json object can contain arbitrary application specific data (json key values). The optional key `backend` is special. It has to conform to the schema documented in `BackendAppData`.
+
+                The total byte size of this field's utf-8 encoded bytes is limited to 1000.
+              type: string
+        appDataHash:
+          description: |
+           May be set for debugging purposes. If set, this field is compared to what the backend internally calculates as the app data hash based on the contents of `appData`. If the hash does not match, an error is returned. If this field is set, then `appData` must be a string encoding of a json object.
```

Comment:
nit: I would start the description by mentioning what it is, rather what you can do with it

Reply:
curious, why is `appDataHash` optional? 
doesn't the user need to provide the signature already that should already be using the matching hash?

Reply:
The api user has three ways to use to provide app data when creating an order:
1. The way it always worked. Providing a hash only. This continues to happen through the field we've always had `appData` for backward compatibility.
2. The new way. By providing the full app data json in the field `appData`.
3. The new way with extra debugging. By providing full app data and this new field `appDataHash`.

If the api user uses approach 2 and makes a mistake when calculating the app data hash locally, then they will get a generic error that signature verification failed. Because the smart contract order they signed had a different app data hash than what the backend used.

In order to make this easier to debug, the api user can provide the hash that they signed too. This allows the backend to produce a better error message.

Reply:
You're right that they already have it. To me it felt cleaner to not be forced to provide it because it can be computed from the full app data. It is redundant. The advantage of being forced to provide it is that we make our code and the api docs simpler. Will think about it.